OMCompiler/Compiler/NFFrontEnd/NFInst.mo

```diff
@@ -229,6 +229,11 @@ end instClassInProgram;
 function resetGlobalFlags
   "Resets the global flags that the frontend uses."
 algorithm
+  if Flags.getConfigBool(Flags.NEW_BACKEND) then
+    FlagsUtil.set(Flags.NF_SCALARIZE, false);
+    FlagsUtil.set(Flags.ARRAY_CONNECT, true);
```

Comment:
I thought we do not want to activate array connect for new backend by default from now on?

Reply:
I just preserved the behaviour we have, just moved it to the frontend instead. What to do with the array connect is still a bit of an open question, we'll see. But it's trivial to remove it anyway.